sample_apps/run_example.sh

```diff
@@ -639,32 +661,60 @@ function run_test() {
 # main() begins here
 # ##################################################################
 
-# Simple command-line arg processing. Expect just one arg, if any.
-if [ $# -eq 1 ]; then
-    if [ "$1" == "--help" ] || [ "$1" == "-h" ]; then
+# Simple command-line arg processing. Expect at least one arg, sample-app-name
+if [ $# -eq 0 ]; then
+    usage_brief
+    exit 1
+fi
+if [ "$1" == "--help" ] || [ "$1" == "-h" ]; then
         usage
         exit 0
-    fi
-    if [ "$1" == "--list" ]; then
-        list_steps
+fi
+
+is_valid_app "$1"
```

Comment:
This validates that the 1st arg is a valid app-name; currently only `simple_app`.